### PR TITLE
Allow filtering PRs by poster in the ListPullRequests API

### PR DESCRIPTION
**Pull request by 6543**
as title

---
*Sponsored by Kithara Software GmbH*

### CODE REVIEW

File: models/issues/pull_list.go [hunk 1]
```diff
@@ -46,6 +47,10 @@ func listPullRequestStatement(ctx context.Context, baseRepoID int64, opts *PullR
 		sess.And("issue.milestone_id=?", opts.MilestoneID)
 	}
 
+	if opts.OwnerID > 0 {
+		sess.And("issue.owner_id=?", opts.OwnerID)
```
Comment (lunny) [inline models/issues/pull_list.go]: I don't think issue has the column `owner_id`.

Comment (6543) [reply models/issues/pull_list.go]: https://github.com/go-gitea/gitea/blob/fa35ace9fb0383cee78e1717bcb6eab1224fb4f1/models/issues/issue.go#L105

Comment (6543) [reply models/issues/pull_list.go]: good catch misslooked while looking at the other api ...

File: routers/api/v1/repo/pull.go [hunk 2]
```diff
@@ -52,63 +52,87 @@ func ListPullRequests(ctx *context.APIContext) {
...
+	//   "500":
+	//     "$ref": "#/responses/error"
 
 	labelIDs, err := base.StringsToInt64s(ctx.FormStrings("labels"))
 	if err != nil {
 		ctx.Error(http.StatusInternalServerError, "PullRequests", err)
 		return
 	}
+	var posterID int64
+	if ctx.FormString("owner") != "" {
```
Comment (6543) [inline routers/api/v1/repo/pull.go]:
```suggestion
	if ctx.FormString("poster") != "" {
```

**Review by lunny — APPROVED** [review]

File: routers/api/v1/repo/pull.go [hunk 3]
```diff
@@ -52,63 +52,87 @@ func ListPullRequests(ctx *context.APIContext) {
...
 	if err != nil {
 		ctx.Error(http.StatusInternalServerError, "PullRequests", err)
 		return
 	}
+	var posterID int64
+	if posterStr := ctx.FormString("poster"); posterStr != "" {
+		poster, err := user_model.GetUserByName(ctx, posterStr)
+		if err != nil {
+			if user_model.IsErrUserNotExist(err) {
+				ctx.Error(http.StatusBadRequest, "Poster not found", err)
```
Comment (delvh) [inline routers/api/v1/repo/pull.go:120]: Hmm… which is better, `400` or `404`?

**Review by delvh — APPROVED** [review]

File: routers/api/v1/repo/pull.go (hunk 3 above)
Comment (6543) [reply routers/api/v1/repo/pull.go:120]: as it is a filter its 400

**Pull request merged**